Chatterino/chatterino2
# Fix tooltip & popup positioning

# Description

This tries to ensure the tooltip & popups are created on the correct
monitor

# Test steps

| | This PR | Main branch |
| - | - | - |
| _Emote tooltip_<br>It shows up on the same screen as my cursor |  |  |
| _Split header tooltip_<br>It shows up on the same screen as my cursor |  |  |
| _Usercard_<br>It shows up on the same screen as my cursor |  |  |
| _Emote popup_<br>It shows up at the same place every time<br>regardless of where my Chatterino or cursor is |  |  |
| REFERENCE | ✔  | ❌ | 

TODO: Add proper test steps to ensure we get testing on all platforms

# Tested on

## Linux Wayland
| UNTESTED | This PR | Main branch |
| - | - | - |
| _Emote tooltip_<br>It shows up on the same screen as my cursor |  |  |
| _Split header tooltip_<br>It shows up on the same screen as my cursor |  |  |
| _Usercard_<br>It shows up on the same screen as my cursor |  |  |
| _Emote popup_<br>It shows up at the same place every time<br>regardless of where my Chatterino or cursor is |  |  |

## Linux X11
| by pajlada on Arch Linux | This PR | Main branch |
| - | - | - |
| _Emote tooltip_<br>It shows up on the same screen as my cursor | ✔ | ❌ |
| _Split header tooltip_<br>It shows up on the same screen as my cursor | ✔ | ❌ |
| _Usercard_<br>It shows up on the same screen as my cursor | ✔ | ❌ |
| _Emote popup_<br>It shows up at the same place every time<br>regardless of where my Chatterino or cursor is | ❌ | ❌ |

## macOS
| UNTESTED | This PR | Main branch |
| - | - | - |
| _Emote tooltip_<br>It shows up on the same screen as my cursor |  |  |
| _Split header tooltip_<br>It shows up on the same screen as my cursor |  |  |
| _Usercard_<br>It shows up on the same screen as my cursor |  |  |
| _Emote popup_<br>It shows up at the same place every time<br>regardless of where my Chatterino or cursor is |  |  |

## Windows 100% zoom
| UNTESTED | This PR | Main branch |
| - | - | - |
| _Emote tooltip_<br>It shows up on the same screen as my cursor | ✔ | ✔ |
| _Split header tooltip_<br>It shows up on the same screen as my cursor | ✔ | ❌ |
| _Usercard_<br>It shows up on the same screen as my cursor | Shows up on the same screen, but doesn't bind to the monitor | Opens up on the wrong screen, but does get bound to the screen size |
| _Emote popup_<br>It shows up at the same place every time<br>regardless of where my Chatterino or cursor is | ✔ | ✔ |

## Windows >100% zoom
| UNTESTED | This PR | Main branch |
| - | - | - |
| _Emote tooltip_<br>It shows up on the same screen as my cursor | ✔ | ✔ |
| _Split header tooltip_<br>It shows up on the same screen as my cursor | ✔ | ❌ |
| _Usercard_<br>It shows up on the same screen as my cursor | Shows up on the same screen, but doesn't bind to the monitor | ❌ |
| _Emote popup_<br>It shows up at the same place every time<br>regardless of where my Chatterino or cursor is | ✔ | ✔ |

Fixes #4733
Fixes #4572
